src/parsers/manifest/metaplaylist/metaplaylist_parser.ts

```diff
@@ -0,0 +1,304 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Manifest, {
+  IAdaptationType,
+  StaticRepresentationIndex,
+  SUPPORTED_ADAPTATIONS_TYPE,
+} from "../../../manifest";
+import idGenerator from "../../../utils/id_generator";
+import {
+  IParsedAdaptation,
+  IParsedAdaptations,
+  IParsedManifest,
+  IParsedPeriod,
+} from "../types";
+import MetaRepresentationIndex from "./representation_index";
+
+export type IAdaptationType = "video"|"audio"|"text"|"image";
+
+export type IParserResponse<T> =
+  {
+    type : "needs-manifest-loader";
+    value : {
+      ressources : Array<{ url : string; transportType : string }>;
+      continue : (loadedRessources : Manifest[]) => IParserResponse<T>;
+    };
+  } |
+  { type : "done"; value : T };
+
+export interface IMetaPlaylistTextTrack {
+  url : string;
+  language : string;
+  closedCaption : boolean;
+  mimeType : string;
+  codecs? : string;
+}
+
+export interface IMetaPlaylist {
+  type : "MPL";
+  version : string;
+  dynamic? : boolean;
+  pollInterval? : number;
+  contents: Array<{
+    url: string;
+    startTime: number;
+    endTime: number;
+    transport: string;
+    textTracks?: IMetaPlaylistTextTrack[];
+  }>;
+}
+
+const generateManifestID = idGenerator();
+
+/**
+ * Parse playlist string to JSON.
+ * Returns an array of contents.
+ * @param {string} data
+ * @param {string} url
+ * @returns {Object}
+ */
+export default function parseMetaPlaylist(
+  data : unknown,
+  url? : string
+): IParserResponse<IParsedManifest> {
+  let parsedData;
+  if (typeof data === "object" && data != null) {
+    parsedData = data;
+  } else if (typeof data === "string") {
+    try {
+      parsedData = JSON.parse(data);
+    } catch (error) {
+      throw new Error("MPL Parser: Bad MetaPlaylist file. Expected JSON.");
+    }
+  } else {
+    throw new Error("MPL Parser: Parser input must be either a string " +
+                    "or the MetaPlaylist data directly.");
+  }
+
+  const { contents, version, type } = parsedData;
+
+  if (type !== "MPL") {
+    throw new Error("MPL Parser: Bad MetaPlaylist. " +
+                    "The `type` property is not set to `MPL`");
+  }
+
+  if (typeof version !== "string" || version.split(".")[0] !== "0") {
```

Comment:
Shouldn't we check that version is precisely `0.1` if every version under the` 1.x `is a breaking version ?

Reply:
Haha. Good catch. Forgot about that.